ros_cross_compile/data_collector.py

```diff
@@ -81,34 +83,59 @@ def __init__(self, ros_workspace_dir: Path,
         self._write_path.mkdir(parents=True, exist_ok=True)
         self.write_file = self._write_path / output_file
 
-    def print_helper(self, data_to_print: List[Dict]):
+    def print_helper(self, data_to_print: List[Datum]):
         print('--------------------------------- Collected Data ---------------------------------')
         print('=================================================================================')
         for datum in data_to_print:
             # readable_time = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(datum['timestamp']))
-            readable_time = datetime.utcfromtimestamp(datum['timestamp']).isoformat()
-            if datum['unit'] == Units.Seconds.value:
-                print('{:>12} | {:>35}: {:.2f} {}'.format(readable_time, datum['name'],
-                                                          datum['value'], datum['unit']),
+            readable_time = datetime.utcfromtimestamp(datum.timestamp).isoformat()
+            if datum.unit == Units.Seconds.value:
+                print('{:>12} | {:>35}: {:.2f} {}'.format(readable_time, datum.metric_name,
+                                                          datum.value, datum.unit),
                       end='')
             else:
-                print('{:>12} | {:>35}: {} {}'.format(readable_time, datum['name'],
-                                                      datum['value'], datum['unit']),
+                print('{:>12} | {:>35}: {} {}'.format(readable_time, datum.metric_name,
+                                                      datum.value, datum.unit),
                       end='')
-            if datum['complete']:
+            if datum.complete:
                 print('\n')
             else:
                 print(' {}'.format('incomplete'))
 
-    def write(self, data_collector: DataCollector, print_data: bool):
+    def serialize_to_cloudwatch(self, data: List[Datum], args: Namespace) -> List[Dict]:
+        """
+        Serialize collected datums to fit the AWS Cloudwatch publishing format.
+
+        To get an idea of what the Cloudwatch formatting is like, refer to
+        https://awscli.amazonaws.com/v2/documentation/api/latest/reference/
+        cloudwatch/put-metric-data.html.
+        """
+        def serialize_helper(datum: Dict) -> Dict:
+            datum['MetricName'] = datum.pop('metric_name')
```

Comment:
create a new dict here, don't change the input - python passes by reference so this function has side effects to the caller. this function would be much more clear as:

```
def serialize_datum(datum: Datum) -> dict:
  return {
    'MetricName': datum.name,
    'Value': datum.value,
    'Unit': datum.unit,
    'Timestamp': datum.timestamp,
    'Dimensions': [
      etc...
    ]
  }
```

Then you can also skip using `_asdict` entirely

```
return [serialize_datum(d) for d in data]
```